// ngx-alerts/esm5/lib/alert.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { AlertComponent } from './component/alert.component';
import { AlertService } from './service/alert.service';
import { CommonModule } from '@angular/common';
import { ALERT_CONFIG } from './alert.config';
var AlertModule = /** @class */ (function () {
    function AlertModule() {
    }
    AlertModule_1 = AlertModule;
    AlertModule.forRoot = function (config) {
        return {
            ngModule: AlertModule_1,
            providers: [
                AlertService,
                { provide: ALERT_CONFIG, useValue: config }
            ]
        };
    };
    var AlertModule_1;
    AlertModule = AlertModule_1 = __decorate([
        NgModule({
            declarations: [
                AlertComponent
            ],
            imports: [
                CommonModule
            ],
            exports: [
                AlertComponent
            ]
        })
    ], AlertModule);
    return AlertModule;
}());
export { AlertModule };
